src/evict/evict_lru.c

```diff
@@ -557,11 +557,17 @@ __evict_update_work(WT_SESSION_IMPL *session)
 	WT_BTREE *las_tree;
 	WT_CACHE *cache;
 	WT_CONNECTION_IMPL *conn;
+	double dirty_target, dirty_trigger, target, trigger;
 	uint64_t bytes_inuse, bytes_max, dirty_inuse;
 
 	conn = S2C(session);
 	cache = conn->cache;
 
+	dirty_target = cache->eviction_dirty_target;
+	dirty_trigger = cache->eviction_dirty_trigger;
+	target = cache->eviction_target;
+	trigger = cache->eviction_trigger;
```

Comment:
This is really the one function that uses these values. My initial thought when I read the config changes would be that instead of computing a `double` as a percentage, why not just convert everything to an absolute value and then I think we can get rid of the multiplications and divisions below and just do comparisons?

Reply:
In case of shared cache the cache size is decided dynamically per connection. In that situation we would want to provide a % input value per connection for these parameters. An absolute value would not be able to adjust like % value per connection. Hence we chose this path of converting absolute to percentage.

Reply:
my understanding is wiredtiger should not take the same % when the cache size is changed. for eg:-
1. wiredtiger open (cache size = 2 TB, eviction target = 1TB)
now eviction target is 50%.
2. reconfigure wiredtiger with cache size = 3 TB, now what will be the eviction target set, will it be 50% of new cache size ( which will be 1.5 TB) or still we consider it as 1 TB ?

my understanding is we have to consider the value provided by user which is 1 TB, and hence when re configured, it should be only 33%  should not be 50%.

Reply:
@bvpvamsikrishna Yes, when a reconfigure is done with a different cache size a new percentage would be calculated based on the absolute value in the input.

But here I am specifically referring the case of shared cache. WiredTiger has a provision to specify a cache pool that can be shared between multiple connections. With shared cache, the cache size is dynamically decided per connection. For a cache size generated dynamically like this, these parameters' (absolute)values should be converted to percentage values, which can be applied to every connection.

Reply:
@sueloverso the reason to keep these as percentages is that otherwise it becomes very difficult to keep the values consistent when the size changes, and in a cache pool configuration it changes regularly. Which made me think about cache pools, and I think we should disallow absolute values here when the cache pool is in use, especially given the special case `don't divide by zero` in the new code for cache pool configurations. There is precedent for similar checks here:
https://github.com/wiredtiger/wiredtiger/blob/develop/src/conn/conn_cache_pool.c#L80
We should add something asserting that the 5 configuration options need to be less than or equal to 100

Reply:
Thanks for the details everyone.